### DISCUSSION TITLE
atomWithReset is always scoped, despite not being passed as dependency to ScopeProvider

### DISCUSSION THREAD

**Discussion by graycodes** [pmndrs/jotai#2661] [Bug report]
### Bug Description

[Link to reproducible issue](https://codesandbox.io/s/modest-curran-rn2fkg?file=/src/App.tsx)

When scoping an `atomWithReset`, it doesn't behave like a normal `atom` does. As you can see from the above sandbox, the `atomWithReset` behaves as if it were included in the `ScopeProvider` atoms dependency array, despite not being in that list. The count increments for each copy of the atom in `<Counter />` independently.

Maybe something about the way that the 'reset' functionality is coded mean that it's being treated as if it is dependent on a scoped atom, or it isn't correctly subscribing to the original atom as suggested [in this post](https://github.com/jotaijs/jotai-scope/issues/25#issuecomment-2014400075)

>  If they are actually not scoped, what we do is like, making the copy subscribe the original atom, so they look like the same. However, internally, all the atoms within a ScopeProvider are copied. They are different instances.

Copy of the code here:
```ts
import { atom, useAtom } from "jotai";
import { ScopeProvider } from "jotai-scope";
import { atomWithReset } from "jotai/vanilla/utils";

const countAtom = atom(0);
const anotherCountAtom = atom(0);
const resettableCountAtom = atomWithReset(0);

const Counter = () => {
  const [count, setCount] = useAtom(countAtom);
  const [anotherCount, setAnotherCount] = useAtom(anotherCountAtom);
  const [resettableCount, setResettableCount] = useAtom(resettableCountAtom);

  return (
    <>
      <div>
        <span>count: {count}</span>
        <button type="button" onClick={() => setCount((c) => c + 1)}>
          increment
        </button>
      </div>
      <div>
        <span>another count: {anotherCount}</span>
        <button type="button" onClick={() => setAnotherCount((c) => c + 1)}>
          increment
        </button>
      </div>
      <div>
        <span>resettable count: {resettableCount}</span>
        <button type="button" onClick={() => setResettableCount((c) => c + 1)}>
          increment
        </button>
      </div>
    </>
  );
};

const App = () => {
  return (
    <div>
      <h1>First Provider</h1>
      <ScopeProvider atoms={[anotherCountAtom]}>
        <Counter />
      </ScopeProvider>
      <h1>Second Provider</h1>
      <ScopeProvider atoms={[anotherCountAtom]}>
        <Counter />
      </ScopeProvider>
    </div>
  );
};

export default App;
```

EDIT: Let me know if this needs to be raised in the jotai-scope repo. Wasn't sure if was a bug here or there.

### Reproduction Link

https://codesandbox.io/s/modest-curran-rn2fkg?file=/src/App.tsx

**Comment by dai-shi** [comment]
@dmaskasky ?

**Comment by dmaskasky** [comment]
## Dependencies
jotai@0.5.0
The issue is still present after upgrading only jotai@0.7.0

jotai-scope@0.4.0
The issue is still present after upgrading jotai-scope@0.7.0

The issue resolved by upgrading both jotai@2.9.0 and jotai-scope@0.7.0.

@dai-shi There is some incompatibility with jotai@0.5.0 and jotai-scope@0.4.0.
